bosh-stemcell/spec/stemcells/fips_spec.rb

```diff
@@ -1,29 +1,155 @@
 require 'spec_helper'
-# TODO: uncomment when fips is ready
-# describe 'FIPS Stemcell', os_image: true do
-#   context 'installed by system_kernel' do
-#     describe package('linux-aws-fips') do
-#       it { should be_installed }
-#     end
-#     describe package('linux-generic-hwe-22.04') do
-#       it { should_not be_installed }
-#     end
-#     describe package('linux-image-5.19.0-109-generic') do
-#       it { should_not be_installed }
-#     end
-#   end
-
-#   context 'installed by base_ssh' do
-#     subject(:sshd_config) { file('/etc/ssh/sshd_config') }
-
-#     it 'allows only secure HMACs' do
-#       macs = %w[
-#           hmac-sha2-512-etm@openssh.com
-#           hmac-sha2-256-etm@openssh.com
-#           hmac-sha2-512
-#           hmac-sha2-256
-#       ].join(',')
-#       expect(sshd_config.content).to match(/^MACs #{macs}$/)
-#     end
-#   end
-# end
+
+describe 'FIPS Stemcell', os_image: true do
+  context 'installed by system_kernel' do
+    describe package('linux-image-fips') do
+      it { should be_installed }
+    end
+    describe package('linux-generic-hwe-22.04') do
+      it { should_not be_installed }
+    end
+    describe package('linux-image-5.19.0-109-generic') do
+      it { should_not be_installed }
+    end
+  end
+
+  context 'installed by base_ssh' do
+    subject(:sshd_config) { file('/etc/ssh/sshd_config') }
+
+    it 'allows only secure HMACs' do
+      macs = %w[
+          hmac-sha2-512-etm@openssh.com
+          hmac-sha2-256-etm@openssh.com
+          hmac-sha2-512
+          hmac-sha2-256
+      ].join(',')
+      expect(sshd_config.content).to match(/^MACs #{macs}$/)
+    end
+
+    it 'enables RSA, ECDSA host keys' do
+      matches = sshd_config.content.scan(/^HostKey.*/)
+
+      expect(matches).to contain_exactly('HostKey /etc/ssh/ssh_host_rsa_key', 'HostKey /etc/ssh/ssh_host_ecdsa_key')
+    end
+  end
+
+  context 'installed by image_install_grub for fips kernel' do
+    describe file('/boot/grub/grub.cfg') do
+      it { should be_file }
+      its(:content) { should match %r{linux\t/boot/vmlinuz-\S+-fips root=UUID=\S* ro } }
+      its(:content) { should match %r{initrd\t/boot/initrd.img-\S+-fips} }
+    end
+  end
+
+  linux_version_regex = 's/linux-(.+)-([0-9]+).([0-9]+).([0-9]+)-([0-9]+)/linux-\1-\2.\3/'
+
+  describe 'installed packages' do
+    dpkg_list_packages = "dpkg --get-selections | cut -f1 | sed -E '#{linux_version_regex}'"
+    # TODO: maby we can use awk "dpkg --get-selections | awk '!/linux-(.+)-([0-9]+.+)/&&/linux/{print $1}'"
```

Comment:
Is this something we want to do?

Reply:
The problem being solved here is not failing our test on every kernel patch:
```
debug/9f7b9093-f357-4397-8a29-67d6d5e7bd56:~$ dpkg --get-selections | cut -f1 | grep linux
binutils-x86-64-linux-gnu
console-setup-linux
libselinux1:amd64
linux-base
linux-fips
linux-fips-headers-5.15.0-73
linux-firmware
linux-headers-5.15.0-73-fips
linux-headers-fips
linux-image-5.15.0-73-fips
linux-image-fips
linux-image-hmac-5.15.0-73-fips
linux-libc-dev:amd64
linux-modules-5.15.0-73-fips
linux-modules-extra-5.15.0-73-fips
```

Becomes:
```
dpkg --get-selections | cut -f1 | sed -E 's/linux-(.+)-([0-9]+).([0-9]+).([0-9]+)-([0-9]+)/linux-\1-\2.\3/' | grep linux
binutils-x86-64-linux-gnu
console-setup-linux
libselinux1:amd64
linux-base
linux-fips
linux-fips-headers-5.15
linux-firmware
linux-headers-5.15-fips
linux-headers-fips
linux-image-5.15-fips
linux-image-fips
linux-image-hmac-5.15-fips
linux-libc-dev:amd64
linux-modules-5.15-fips
linux-modules-extra-5.15-fips
util-linux
```

The proposed `awk` based solution (which can not be used as is because it filters the list down to only packages with linux in their name) proposes to drop versions from linux packages all together:
```
sudo dpkg --get-selections | awk '!/linux-(.+)-([0-9]+.+)/&&/linux/{print $1}'
binutils-x86-64-linux-gnu
console-setup-linux
libselinux1:amd64
linux-base
linux-fips
linux-firmware
linux-headers-fips
linux-image-fips
linux-libc-dev:amd64
util-linux
```

I'm not a fan of removing versions altogether and would like our tests to still fail on Minor kernel bumps. 
If someone feels like the current behavior could be implemented with `awk` in a better way, be my guest, but this should not block this PR.

Reply:
yeah i wrote this `#todo` before we where solving it with the `update_kernel_static_libraries` function.
i think we can remove this todo comment it has been there for a few years now